graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,45 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly."""
```

Comment:
Consider a summary of the rules in the top-level docstring, it takes a little digging to figure out what's allowed for directives otherwise.